src/webgpu/api/operation/render_pass/storeOp.spec.ts

```diff
@@ -1,23 +1,352 @@
+    const storeOps: GPUStoreOp[] = [];
+    for (let i = 0; i < t.params.colorAttachments; i++) {
+      if (i % 2 === 0) {
+        storeOps.push(t.params.storeOperation1);
+      } else {
+        storeOps.push(t.params.storeOperation2);
+      }
+    }
+
+    // Color load operation will clear to {1.0, 0.0, 0.0, 1.0}
+    // Color store operation is determined by test params.
+    const renderPassColorAttachmentDescriptors: GPURenderPassColorAttachmentDescriptor[] = [];
+    for (let i = 0; i < t.params.colorAttachments; i++) {
+      renderPassColorAttachmentDescriptors.push({
+        attachment: colorAttachmentViews[i],
+        loadValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+        storeOp: storeOps[i],
+      });
+    }
+
+    const encoder = t.device.createCommandEncoder();
+    const pass = encoder.beginRenderPass({
+      colorAttachments: renderPassColorAttachmentDescriptors,
+    });
+    pass.endPass();
+    t.device.defaultQueue.submit([encoder.finish()]);
+
+    // Check that the correct store operation occurred.
+    let expectedValue: PerTexelComponent<number> = {};
+    for (let i = 0; i < t.params.colorAttachments; i++) {
+      if (storeOps[i] === 'clear') {
+        // If colorStoreOp was clear, the texture should now contain {0.0, 0.0, 0.0, 0.0}.
+        expectedValue = { R: 0.0, G: 0.0, B: 0.0, A: 0.0 };
+      } else if (storeOps[i] === 'store') {
+        // If colorStoreOP was store, the texture should still contain {1.0, 0.0, 0.0, 1.0}.
+        expectedValue = { R: 1.0, G: 0.0, B: 0.0, A: 1.0 };
+      }
+      t.expectSingleColor(colorAttachments[i], kColorFormat, {
+        size: [kHeight, kWidth, 1],
+        exp: expectedValue,
+      });
+    }
+  });
+
+// Tests that render pass depth stencil store operations work correctly for all renderable color
+// formats, mip levels and array layers.
+g.test('render_pass_store_op,depth_stencil_attachment_only')
+  .params(
+    params()
+      .combine(poptions('depthStencilFormat', kTextureFormats))
+      // Filter out color and non-renderable formats.
+      .unless(
+        ({ depthStencilFormat }) =>
+          kTextureFormatInfo[depthStencilFormat].color ||
+          !kTextureFormatInfo[depthStencilFormat].renderable ||
+          !kTextureFormatInfo[depthStencilFormat].copyable
+      )
+      .combine(poptions('storeOperation', kStoreOps))
+      .combine(poptions('mipLevel', kMipLevel))
+      .combine(poptions('arrayLayer', kArrayLayers))
+  )
+  .fn(t => {
+    const depthStencilAttachment = t.device.createTexture({
+      format: t.params.depthStencilFormat,
+      size: { width: kWidth, height: kHeight, depth: t.params.arrayLayer + 1 },
+      mipLevelCount: t.params.mipLevel + 1,
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.OUTPUT_ATTACHMENT,
+    });
+
+    const depthStencilViewDesc: GPUTextureViewDescriptor = {
+      format: t.params.depthStencilFormat,
+      dimension: '2d',
+      aspect: 'all',
```

Comment:
We should probably test rendering into views with subsets of the aspects to make sure those also behave properly.

Reply:
I agree - although Dawn doesn't support anything other than "all" at the moment, which will make it difficult for me to test my test.

Reply:
sure, let's have a TODO for this

Reply:
TODO added.